src/liballoc/boxed.rs

```diff
@@ -596,6 +596,40 @@ impl From<Box<str>> for Box<[u8]> {
     }
 }
 
+use core::convert::TryFrom;
+
+#[derive(Debug, Copy, Clone)]
+#[unstable(feature = "boxed_slice_try_from", issue = "0")]
+/// Can we use TryFromSliceError instead?
+/// TODO: implement Error
+pub struct PlaceholderError(());
```

Comment:
We may want to create a custom error in order to recover the original allocation in case of failure.